File: Algorithms/Implementation/Jumping on the Clouds: Revisited.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class Solution {

    // Complete the jumpingOnClouds function below.
    static int jumpingOnClouds(int[] c, int k) {

        int n = c.Length;
        int energy = 100;
        int i = 0;

        while (true)
        {
            i += k;
            if (i >= n) i %= n;

            energy -= c[i] == 1 ? 2 : 0;
            energy--;

            if (i % n == 0) break;
        }
        return energy;
    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string[] nk = Console.ReadLine().Split(' ');

        int n = Convert.ToInt32(nk[0]);
        int k = Convert.ToInt32(nk[1]);

        int[] c = Array.ConvertAll(Console.ReadLine().Split(' '), cTemp => Convert.ToInt32(cTemp))
        ;
        int result = jumpingOnClouds(c, k);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Implementation/Jumping on the Clouds: Revisited.php
<?php

# Complete the jumpingOnClouds function below.
function jumpingOnClouds($c, $k) {
    $n = count($c);
    $energy = 100;
    $i = 0;
    while (true) {
        $energy -= 1;
        $i = ($i + $k) % $n;
        if ($c[$i] == 1) {
            $energy -= 2;
        }
        if ($i == 0) {
            break;
        }
    }

    return $energy;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%[^\n]", $nk_temp);
$nk = explode(' ', $nk_temp);

$n = intval($nk[0]);
$k = intval($nk[1]);

fscanf($stdin, "%[^\n]", $c_temp);

$c = array_map('intval', preg_split('/ /', $c_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = jumpingOnClouds($c, $k);

fwrite($fptr, $result . "\n");

fclose($stdin);
fclose($fptr);
